pkg/controller/database/postgresqlserver/managed.go

```diff
@@ -146,6 +149,23 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	return o, nil
 }
 
+func (e *external) getPassword(ctx context.Context, cr *v1beta1.PostgreSQLServer) (string, error) {
+	if cr == nil || cr.Spec.WriteConnectionSecretToReference == nil ||
```

Comment:
```suggestion
	if cr.Spec.WriteConnectionSecretToReference == nil ||
```

nitpick: we'd need to place nil guard to just too many spaces if we were to check existence of the CR. So, we usually assume it's there after the classic `cr, ok := mg.(*v1beta1.PostgreSQLServer)` statement.

Reply:
Done.